tests/unit/test_call_endpoint.py

```diff
@@ -0,0 +1,216 @@
+  - Python 2.7 (ActivePython recommended on Windows)
+  - Python 3.5 (ActivePython recommended on Windows)
+
+"""
+
+import os
+import unittest
+import siem
+
+import mock
+import json
+
+
+class CallEndPointTest(unittest.TestCase):
+
+    @mock.patch('siem.config.Token')
+    @mock.patch('siem.store_state')
+    def test_data_stream_in_event(self,
+                                  mock_config_token,
+                                  mock_store_state
+                                  ):
+        # Setup
+        # Sample event
+        mock_event_response = {
+            "has_more": False,
+            "next_cursor": "VjJfQ1VSU09SfDIwMTktMDQtMDFUMTg6MjU6NDEuNjA2Wg==",
+            "items": [
+                {
+                    "when": "2019-04-01T15:11:09.759Z",
+                    "id": "cbaff14f-a36b-46bd-8e83-6017ad79cdef",
+                    "customer_id": "816f36ee-dd2e-4ccd-bb12-cea766c28ade",
+                    "severity": "low",
+                    "created_at": "2019-04-01T15:11:09.984Z",
+                    "source_info": {
+                        "ip": "10.1.39.32"
+                    },
+                    "endpoint_type": "server",
+                    "endpoint_id": "c80c2a87-42f2-49b2-bab7-5031b69cd83e",
+                    "origin": None,
+                    "type": "Event::Endpoint::Registered",
+                    "location": "mock_Mercury_1",
+                    "source": "n/a",
+                    "group": "PROTECTION",
+                    "name": "New server registered: mock_Mercury_1"
+                },
+                {
+                    "when": "2019-04-01T15:11:41.000Z",
+                    "id": "5bc48f19-3905-4f72-9f79-cd381c8e92ce",
+                    "customer_id": "816f36ee-dd2e-4ccd-bb12-cea766c28ade",
+                    "severity": "medium",
+                    "created_at": "2019-04-01T15:11:41.053Z",
+                    "source_info": {
+                        "ip": "10.1.39.32"
+                    },
+                    "endpoint_type": "server",
+                    "endpoint_id": "c80c2a87-42f2-49b2-bab7-5031b69cd83e",
+                    "origin": None,
+                    "type": "Event::Endpoint::Threat::Detected",
+                    "location": "mock_Mercury_1",
+                    "source": "n/a",
+                    "group": "MALWARE",
+                    "name": "Malware detected: 'Eicar-AV-Test' at 'C:\\Program Files (x86)\\Trojan Horse\\bin\\eicar.com'"
+                }
+            ]
+        }
+
+        events = []
+        # Run
+        try:
+            with mock.patch('siem.request_url') as mock_request_url:
+                mock_request_url.return_value = json.dumps(mock_event_response)
+                for e in siem.call_endpoint(mock.Mock(), '/siem/v1/events', False,
+                           False, 'fake_state_file', mock_config_token):
+                    events.append(e)
+
+        except Exception as ex:
+            print ex
```

Comment:
This could use some more descriptive text, but this isn't a blocker.

Reply:
Done